arch/arm/core/Kconfig

```diff
@@ -69,9 +69,7 @@ endchoice
 
 endmenu
 
-if CPU_CORTEX_M
```

Comment:
Won't this expose configs like `CPU_CORTEX_M_HAS_BASEPRI` and `CPU_CORTEX_M_HAS_VTOR` to non-Cortex-M arches? Should we move them into the "ARM Cortex-M options" menu?

Reply:
Or, perhaps keep the conditional here, as follows:

`if CPU_CORTEX_M0 || CPU_CORTEX_M0PLUS || CPU_CORTEX_M3 || CPU_CORTEX_M4 || CPU_CORTEX_M23 || CPU_CORTEX_M33 || CPU_CORTEX_M7`

I know, this is a bit ugly, but it is not likely that we have to touch this line too often.

I think that we need to keep the conditional "source", for cortex_m/Kconfg; we should try not to expose any of the K-options defined there, if we don't have a Cortex-M CPU.